openssl-dynamic/src/main/c/sslutils.c

```diff
@@ -329,37 +327,51 @@ static unsigned char dhxxx2_g[]={
 static DH *get_dh(int idx)
 {
     DH *dh;
-    if ((dh = DH_new()) == NULL)
+    BIGNUM *p, *g;
+    if ((dh = DH_new()) == NULL) {
         return NULL;
+    }
+
     switch (idx) {
         case SSL_TMP_KEY_DH_512:
-            dh->p = BN_bin2bn(dh0512_p, sizeof(dh0512_p), NULL);
+            p = BN_bin2bn(dh0512_p, sizeof(dh0512_p), NULL);
         break;
         case SSL_TMP_KEY_DH_1024:
-            dh->p = BN_bin2bn(dh1024_p, sizeof(dh1024_p), NULL);
+            p = BN_bin2bn(dh1024_p, sizeof(dh1024_p), NULL);
         break;
         case SSL_TMP_KEY_DH_2048:
-            dh->p = BN_bin2bn(dh2048_p, sizeof(dh2048_p), NULL);
+            p = BN_bin2bn(dh2048_p, sizeof(dh2048_p), NULL);
         break;
         case SSL_TMP_KEY_DH_4096:
-            dh->p = BN_bin2bn(dh4096_p, sizeof(dh2048_p), NULL);
+            p = BN_bin2bn(dh4096_p, sizeof(dh2048_p), NULL);
         break;
     }
-    dh->g = BN_bin2bn(dhxxx2_g, sizeof(dhxxx2_g), NULL);
-    if ((dh->p == NULL) || (dh->g == NULL)) {
-        DH_free(dh);
-        return NULL;
+    if (p == NULL) {
```

Comment:
Could `p` be uninitialized here?

Reply:
yes `BN_bin2bn` will return `NULL` on error... https://www.openssl.org/docs/man1.1.0/crypto/BN_bin2bn.html

Reply:
But there is no `default` case in the switch.

Reply:
good point... let me just init it above.